test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -6858,17 +6858,22 @@ TEST_P(CdsTest, RingHashNoHashPolicy) {
   gpr_setenv("GRPC_XDS_EXPERIMENTAL_ENABLE_RING_HASH", "true");
   const double kDistribution50Percent = 0.5;
   const double kErrorTolerance = 0.05;
+  const uint32_t kRpcTimeoutMs = 5000;
```

Comment:
Why does increasing the ring size require increasing the timeouts?  The two endpoints are both up, so neither of them should cause delays from trying to connect.

Same thing for all of these tests.

Reply:
It takes seconds to create the Picker for ring hash policy, and we do that multiple times: idle->connecting, connecting->ready, and for multiple channel changes
I0517 06:37:57.427480420      20 ring_hash.cc:343]           donna starting to create the picker: the ring
I0517 06:37:59.829846291      20 ring_hash.cc:376]           [RH 0x7b14000014f0 picker 0x7b0c00017e20] created picker from subchannel_list=0x7b7400012200 with 100000 ring entries

So the very first RPC can timeout

Reply:
Woah -- that's surprising.  Why is it taking so long to create the picker?

As evidenced by this test, we create new pickers whenever any one individual subchannel changes state, and that can happen pretty frequently, so picker creation really needs to be fast.  If it's taking this long, then we need to find a way to optimize it.